rocketpy/sensors/sensors.py

```diff
@@ -8,6 +9,265 @@
+        self.temperature_bias = temperature_bias
+        self.temperature_scale_factor = temperature_scale_factor
+        self.name = name
+        self.measurement = None
+        self.measured_data = []
+        self._counter = 0
+        self._save_data = self._save_data_single
+        self._random_walk_drift = 0
+        self.normal_vector = Vector([0, 0, 0])
+
+        # handle measurement range
+        if isinstance(measurement_range, (tuple, list)):
+            if len(measurement_range) != 2:
+                raise ValueError("Invalid measurement range format")
+            self.measurement_range = measurement_range
+        elif isinstance(measurement_range, (int, float)):
+            self.measurement_range = (-measurement_range, measurement_range)
+        else:
+            raise ValueError("Invalid measurement range format")
+
+        # map which rocket(s) the sensor is attached to and how many times
+        self._attached_rockets = {}
+
+    def __repr__(self):
+        return f"{self.name}"
+
+    def __call__(self, *args, **kwargs):
+        return self.measure(*args, **kwargs)
+
+    def _reset(self, simulated_rocket):
+        """Reset the sensor data for a new simulation."""
+        self._random_walk_drift = (
+            Vector([0, 0, 0]) if isinstance(self._random_walk_drift, Vector) else 0
+        )
+        self.measured_data = []
+        if self._attached_rockets[simulated_rocket] > 1:
+            self.measured_data = [
+                [] for _ in range(self._attached_rockets[simulated_rocket])
+            ]
+            self._save_data = self._save_data_multiple
+        else:
+            self._save_data = self._save_data_single
+
+    def _save_data_single(self, data):
+        """Save the measured data to the sensor data list for a sensor that is
+        added only once to the simulated rocket."""
+        self.measured_data.append(data)
+
+    def _save_data_multiple(self, data):
+        """Save the measured data to the sensor data list for a sensor that is
+        added multiple times to the simulated rocket."""
+        self.measured_data[self._counter].append(data)
+        # counter for cases where the sensor is added multiple times in a rocket
+        self._counter += 1
+        if self._counter == len(self.measured_data):
+            self._counter = 0
+
+    @abstractmethod
+    def measure(self, time, **kwargs):
+        pass
+
+    @abstractmethod
+    def export_measured_data(self):
+        pass
+
+    @abstractmethod
+    def quantize(self, value):
+        """Quantize the sensor measurement"""
+        pass
+
+    @abstractmethod
+    def apply_noise(self, value):
+        """Add noise to the sensor measurement"""
+        pass
+
+    @abstractmethod
+    def apply_temperature_drift(self, value):
+        """Apply temperature drift to the sensor measurement"""
+        pass
+
+    def export_measured_data(self, filename, format, data_labels):
+        """
```

Comment:
method already defined in line 189 PylintE0102:function-redefined

`exoport_measured_data`